src/Structs.vala

```diff
@@ -34,7 +34,19 @@ namespace FeedReader {
 		}
 	}
 
-	private struct ResourceMetadata
+	public struct FavIconData
+	{
+		public Gdk.Pixbuf? Icon;
+		public ResourceMetadata? Metadata;
+
+		public FavIconData(Gdk.Pixbuf? icon, ResourceMetadata? meta)
+		{
+			Metadata = meta;
+			Icon = icon;
+		}
+	}
```

Comment:
This should probably be defined in `FavIconManager` since it doesn't have any use outside of that class.